### ISSUE TITLE
Manage integrations

### ISSUE THREAD

**Issue by olimart** [yafoy/pablo#1]
- Build mechanism to activate/deactivate pre-approved integrations (integrations#index)
  - Integrations will separate gem (for example lita-slack), so need some kind catalogue to list approved integrations. Integrations must be installed only if user decides so. Need to think about updating integrations as well, still from the UI  
  - For now, let's focus only on Slack as adapter
- System must ask user for configuration details. Slack robot token for example.
  - Token needs to be saved to db
- Need to think about packaging the app. I'm planning to make it easy to distribute, bootstrap and upgrade. Bootstrapping a rails app is easy but here there will be dependencies (adapters, plugins) that will need to be installed on-demand. So I'm planning to have "approved" plugins that the user can configure anytime. Think about the way Slack integrations works for example
Also I know that Lita will crash if no adapter set initially and obviously we don't want that so we need to ensure that user completes setup.
- Tests

**Comment by adbjesus** [comment]
@olimart From what I understand the pre-approved integrations are those installed with the corresponding gem (e.g. pablo-slack) when the user starts the bot. So for example let's say I have 2 integrations
* pablo-slack
* pablo-hipchat
but only the pablo-slack gem is installed in my system. Therefore, in the integrations list, only the slack integration would show up. Is this correct? Or is the behaviour different?

**Comment by olimart** [comment]
> Therefore, in the integrations list, only the slack integration would show up. Is this correct? Or is the behaviour different?

both would appear. Up to the user to choose which ones to activate. Each integrations should have status (active/inactive). For now let's assume that the catalogue of pre-approved integrations are served locally. In future, they will most likely served trough some web service. Hope this helps.

We also need to separate integrations into categories:

- Adapters: slack, hip chat...
- "business-related" integrations: google image...

**Comment by adbjesus** [comment]
So, the platform would have to be able to install the gem if it isn't previously installed or assume all the gems are installed by default (e.g. pablo-hipchat above wasn't installed, if we wanted to activate it, we must install the gem first)?

Ok, about the adapters/categories, and it being served locally.

**Comment by olimart** [comment]
> So, the platform would have to be able to install the gem if it isn't previously installed or assume all the gems are installed by default (e.g. pablo-hipchat above wasn't installed, if we wanted to activate it, we must install the gem first)?

correct. no gems come installed (so we can guarantee that the latest version gets installed), therefore when you click 'activate' download begins.

I think actually installation begins when user provides credentials (and system validates).
